### PR TITLE
Update build versioning docs

### PR DESCRIPTION
**Pull request by craastad**
Update build versioning docs to include a TLDR about App Store and Google Play versioning requirements, Flutter versioning, and sensible suggestions of what to do. The goal is to educate users on build versioning so they can sensibly debug their own issues.

### CODE REVIEW

File: content/building/build-versioning.md [hunk 1]
```diff
@@ -4,25 +4,59 @@ description: How to set a new build number to push to app stores
...
+
+[CFBundleVersion](https://developer.apple.com/documentation/bundleresources/information_property_list/cfbundleversion), Build Version Number, is the internal build version number of your application used for testing and development. It appears in `{major}.{minor}.{patch}` format of one to three period separated integers. If {minor}.{patch} are not provided, then they will default to zero. Build version number must be incremented with every release candidate submitted to test flight for a particular release version number. For iOS, build version number can be reused across different release version numbers while for macOS, build version number must be unique across all release version numbers. This value is best incremented and set by your CI/CD pipeline for every build your submitting to TestFlight.
```
Comment (helinanever) [inline content/building/build-versioning.md:17]:
```suggestion
[CFBundleVersion](https://developer.apple.com/documentation/bundleresources/information_property_list/cfbundleversion), Build Version Number, is the internal build version number of your application used for testing and development. It appears in `{major}.{minor}.{patch}` format of one to three period separated integers. If {minor}.{patch} are not provided, then they will default to zero. Build version number must be incremented with every release candidate submitted to TestFlight for a particular release version number. For iOS, build version number can be reused across different release version numbers while for macOS, build version number must be unique across all release version numbers. This value is best incremented and set by your CI/CD pipeline for every build you're submitting to TestFlight.
```

File: content/building/build-versioning.md [hunk 2]
```diff
@@ -4,25 +4,59 @@ description: How to set a new build number to push to app stores
...
+
+### Google Play Requirements
+
+You can find the Google Play build versioning requirements in the [Android documentation](https://developer.android.com/studio/publish/versioning#appversioning). The important values defined in the build.gradle file are `versionName` and `versionCode`.
+
+`versionName` is a text based, external, version of your app that is displayed to users and visible in Google Play. There are no restrictions for `versionName`, so you should choose something that makes sense for you and your users, such as {major}.{minor}.{patch} versioning. It's advisable to commit this value to version control and update it for every new release of your app to the Play Store.
```
Comment (helinanever) [inline content/building/build-versioning.md:23]:
```suggestion
`versionName` is a text based, external, version of your app that is displayed to users and visible in Google Play. There are no restrictions for `versionName`, so you should choose something that makes sense for you and your users, such as `{major}.{minor}.{patch}` versioning. It's advisable to commit this value to version control and update it for every new release of your app to the Play Store.
```

**Comment by helinanever** [comment]
Great stuff, @craastad !

**Comment by craastad** [comment]
It would be great if @VeArnold could take a look :)

**Review by VeArnold — APPROVED** [review]

**Comment by VeArnold** [comment]
LGTM

**Pull request merged**